src/telemetry/ITelemetry.ts

```diff
@@ -0,0 +1,7 @@
+import { ComponentType } from 'react';
+
+export default interface ITelemetry {
+  initialize(): void;
+  collect(eventName: string, payload: any): void;
```

Comment:
You have trackComponent and trackException. I propose trackEvent instead of *collect()*. What other types of data might we capture? trackMetrics?